src/main/webapp/scripts/group-script.js

```diff
@@ -22,15 +28,147 @@ function getAllGroups(){
+        .attr("pointer-events", d => !d.children ? "none" : null)
+        .on("mouseover", function() { d3.select(this).attr("stroke", "#000"); })
+        .on("mouseout", function() { d3.select(this).attr("stroke", null); })
+        .on("click", d => focus !== d && (zoom(d), d3.event.stopPropagation()));
+
+    const label = svg.append("g")
+        .style("font", "10px sans-serif")
+        .attr("pointer-events", "none")
+        .attr("text-anchor", "middle")
+        .selectAll("text")
+        .data(root.descendants())
+        .join("text")
+        .style("fill-opacity", d => d.parent === root ? 1 : 0)
+        .style("display", d => d.parent === root ? "inline" : "none")
+        .text(d => d.data.name);
+
+
+    zoomTo([root.x, root.y, root.r * 2]);
+
+    function zoomTo(v) {
+        const k = width / v[2];
+
+        view = v;
+
+        label.attr("transform", d => `translate(${(d.x - v[0]) * k},${(d.y - v[1]) * k})`);
+        node.attr("transform", d => `translate(${(d.x - v[0]) * k},${(d.y - v[1]) * k})`);
+        node.attr("r", d => d.r * k);
+    }
+
+    function zoom(d) {
+        const focus0 = focus;
+
+        focus = d;
+
+        const transition = svg.transition()
+            .duration(d3.event.altKey ? 7500 : 750)
+            .tween("zoom", d => {
+            const i = d3.interpolateZoom(view, [focus.x, focus.y, focus.r * 2]);
+            return t => zoomTo(i(t));
+            });
+
+        label
+        .filter(function(d) { return d.parent === focus || this.style.display === "inline"; })
+        .transition(transition)
+            .style("fill-opacity", d => d.parent === focus ? 1 : 0)
+            .on("start", function(d) { if (d.parent === focus) this.style.display = "inline"; })
+            .on("end", function(d) { if (d.parent !== focus) this.style.display = "none"; });
+    }
+
+    var chartElement = document.getElementById("chart")
+    chartElement.appendChild(svg.node())
+
+    return svg.node();
 }
 
 /* Create each of the group <div> elements and display them */
-function loadGroups(groups) {
-    let groupsContainer = document.getElementById("groups");
+async function loadGroups(groups) {
+    var groupsContainer = document.getElementById("groups");
+    // retrieve members from each group
     for (var i = 0; i < groups.length; i++) {
         groupsContainer.appendChild(createGroupElement(groups[i]));
+        // create a new data point in the global data object for this group
+        var newCircle = {
+            name: groups[i].name,
+            children: []
+        }
+
+        const response = await fetch('https://www.googleapis.com/admin/directory/v1/groups/' + groups[i].id + '/members', {
+        headers: {
+            'authorization': `Bearer ` + token,
+        }
+        })
+        const json = await response.json();
+        console.log(json)
+
+        if (response.status == 200) {
+            var members = json.members;
+            var allMembers = true;
```

Comment:
I think the variable should be renamed, it seem to be true of the group has no nested groups, maybe something like containsSubGroups